system-parachains/bridge-hubs/bridge-hub-kusama/src/lib.rs

```diff
@@ -472,6 +507,179 @@ impl pallet_utility::Config for Runtime {
 	type WeightInfo = weights::pallet_utility::WeightInfo<Runtime>;
 }
 
+// Ethereum Bridge
+parameter_types! {
+	pub storage EthereumGatewayAddress: H160 = H160(hex_literal::hex!("EDa338E4dC46038493b885327842fD3E301CaB39"));
```

Comment:
Perhaps very DQ, but I'm just starting here :) Gateway address is the same at Kusama and Polkadot BH. Is that correct? If we'll be using the same gateway contract, can the message be dispatched on both bridge hubs? Looking at the inbound pallet code:
```rust
			// submit message to verifier for verification
			T::Verifier::verify(&message.event_log, &message.proof)
				.map_err(|e| Error::<T>::Verification(e))?;

			// Decode event log into an Envelope
			let envelope =
				Envelope::try_from(&message.event_log).map_err(|_| Error::<T>::InvalidEnvelope)?;

			// Verify that the message was submitted from the known Gateway contract
			ensure!(T::GatewayAddress::get() == envelope.gateway, Error::<T>::InvalidGateway);

			// Retrieve the registered channel for this message
			let channel =
				T::ChannelLookup::lookup(envelope.channel_id).ok_or(Error::<T>::InvalidChannel)?;
```

To me it looks like the same message could pass all the checks if lookup will return a channel (and IIUC it willl, because some channel ids are static and others are derived based on para id, which is the same on Kusama and Polkadot).

Reply:
Actually Gateway address will be different between Kusama and Polkadot. Check here:
```
// Verify that the message was submitted from the known Gateway contract
ensure!(T::GatewayAddress::get() == envelope.gateway, Error::<T>::InvalidGateway);
```

Reply:
Ok, good to know that it should be different :) But now it is the same on both BHs. Is it going to be changed later, using governance calls? Then is it safe to set it before that in code to some `EDa338E4dC46038493b885327842fD3E301CaB39` (which looks like something copied from Rococo configuration). Maybe let's set it to some dummy value then? Or if you know gateway address in advance, then please fix it here

Reply:
Yeah, we should initialize the address to `H160::zero()` here.